### PR TITLE
[APM2-59] fix underline issue related to disable merchant new order email for on-hold

### PR DESCRIPTION
**Pull request by nimitomise**
1. Objective

- Fix underline issue related to disable merchant new order email for on-hold.
- Send notification email to merchant once change order status to processing depending on payment method.

Related information:
Related PR(s): #232

2. Description of change

Override Woocommerce filter for woocommerce_email_recipient_new_order which able to initiate with status as on-hold to not send order email to merchant.
Override woocommerce_order_status_processing to send notification email depend on payment method

3. Quality assurance

- no email send to merchant on complete checkout screen
- found single email send to merchant when change order status to `processing`
- new order email to customer both status as on-hold and processing

🔧 Environments:

Tested locally by pointing OMISE_API_URL to staging-omise

WooCommerce: v5.7.1
WordPress: v5.8
PHP version: 7.1

✏️ Details:

Make sure setting merchant email at /wp-admin/admin.php?page=wc-settings&tab=email&section=wc_email_new_order or /wp-admin/admin.php?page=wc-settings&tab=email to your valid email
Checkout cart with Paynow payment make sure customer email is valid and accessible
Go to customer mail box should still receive new order email
Go to merchant mail box should not receive new order email
Checkout cart with Credit Card and accessible
Go to customer mail box should still receive new order email
Go to merchant mail box should still receive new order email

4. Impact of the change
No

5. Priority of change
Normal

6. Additional Notes
No

### CODE REVIEW

**Comment by nimitomise** [comment]
> just checking if you've tested other payment methods to see they still work as expected 🙂
Yes. I have checked checkout with internet banking then an email with processing status to customer and an email to merchant.

File: includes/gateway/class-omise-payment.php [hunk 1]
```diff
@@ -74,16 +74,38 @@ abstract class Omise_Payment extends WC_Payment_Gateway {
...
+	 * but need to resend for other payment method otherwise merchant will get duplicate email once new order created
+	 * @param string $order_id
+	 * @param string|WC_Order $order
+	 */
+	public function email_processing_admin_notification( $order_id, $order ) {
+		$payment_gateway = wc_get_payment_gateway_by_order( $order );
+		if (is_a( $payment_gateway, 'Omise_Payment' ) && $payment_gateway->enabled_processing_notification) {
+			WC()->mailer()->get_emails()['WC_Email_New_Order']->trigger( $order_id );
+		}
+	}
```
Comment (kan98) [inline includes/gateway/class-omise-payment.php:99]: not sure if it's worth refactoring this into a helper or service class. Benefit is we can move this logic out of this class and we can write some unit tests and mock a few orders to see they work as expected.

Comment (nimitomise) [reply includes/gateway/class-omise-payment.php:99]: Let's me see

File: includes/libraries/omise-plugin/helpers/mailer.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+         * @param string $recipient
+         * @param string|WC_Order $order
+         * @return mixed|string
+         */
+        public function disable_merchant_order_on_hold( $recipient, $order ) {
+            if (is_a($order, 'WC_Order') && $order->get_status() == 'on-hold' ) $recipient = '';
+            return $recipient;
+        }
+    }
+}
```
Comment (kan98) [inline includes/libraries/omise-plugin/helpers/mailer.php:28]: EOF

**Comment by kan98** [comment]
> Try to add unit test for Helper function but too complicated !!

No problem! Thanks for refactoring the function out 🙂

**Pull request merged**